torch_xla/experimental/xla_sharding.py

```diff
@@ -412,35 +402,24 @@ def mark_sharding(t: Union[torch.Tensor, XLAShardedTensor], mesh: Mesh,
   assert len(specs) == len(np.unique(specs)), \
     f"Each device mesh dimension should appear at most once in partition_spec {partition_spec}."
 
-  # check for sharding 2D tensor on a 3D mesh
-  original_shape = tuple(t.shape)
-  # number of dims to expand on tensor
-  tensor_expand = 0
-  if tensor_expand < len(mesh.get_logical_mesh().shape) - len(partition_spec):
-    tensor_expand = len(mesh.get_logical_mesh().shape) - len(partition_spec)
-    partition_spec = (None,) * tensor_expand + partition_spec
-    shape = (1,) * tensor_expand + (*original_shape,)
-    t = t.expand(shape)
-
   tile_assignment = _get_tile_assignment(mesh, partition_spec)
-  sharding_type = _get_sharding_type(partition_spec, num_devices)
+  mesh_size = 1 if type(mesh.mesh_shape) == int else len(mesh.mesh_shape)
```

Comment:
Good question, single element int tuple could be passed as int.

Reply:
Single-element tuples should be declared as e.g. `(1,)` instead of `(1)`. I'm concerned about having to special-case for this, do we get much benefit from supporting both paradigms?

Reply:
We would need to handle it in other places too, like https://github.com/pytorch/xla/pull/5411/files#diff-66621cfc79f7e139b195eaee8de1df71b4870d57294fd8e416a2e7a597bea0b0R340

Reply:
The input is type checkes `:Tuple` and should be passed as `(1,)`. I am also surprised to see that it's sometimes interpreted as int. Let me check.

Reply:
Ok, found the bug in `test_mark_sharding_ir_with_multiple_output`, whcih caused the 
```
TypeError: object of type 'int' has no len()
```
error.

Reply:
https://github.com/pytorch/xla/pull/5411/files#r1286401442